Demo/Demo/PayPalWebPayments/PayPalWebViewModel/PayPalWebViewModel.swift

```diff
@@ -100,47 +154,41 @@ class PayPalWebViewModel: ObservableObject {
             payPalDataCollector = PayPalDataCollector(config: config)
             return payPalClient
         } catch {
-            updateState(.error(message: error.localizedDescription))
+            DispatchQueue.main.async {
+                self.state.createdOrderResponse = .error(message: error.localizedDescription)
+            }
             print("❌ failed to create PayPalWebCheckoutClient with error: \(error.localizedDescription)")
             return nil
         }
     }
 
     func completeTransaction() async throws {
         do {
-            updateState(.loading)
-
-            let payPalClientMetadataID = payPalDataCollector?.collectDeviceData()
-            if let orderID {
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .loading
```

Comment:
you need to check the intent and if it's authorize, then use this state, otherwise use the capturedOrderResponse